File: solutions/go/etl/1/etl.go

// package etl will transform a scrabble scoring system
package etl
import "strings"

// Transform fill transform a map of value, letters to a map of letters, value
func Transform(in map[int][]string) map[string]int {
	result := make(map[string]int)

	for value, letters := range in {
		for _,letter := range letters {
			result[strings.ToLower(letter)] = value
		}
	}
	return result
}

File: solutions/cpp/etl/1/etl.cpp
#include "etl.h"
using namespace std;
namespace etl {
   map<char, int> transform(map<int, std::vector<char>> old){
       map<char, int> newdic;
       for (auto const& x : old) {
           auto score = x.first;
           auto chs = x.second;
           for (auto const& y : chs) {
               newdic[tolower(y)]=score;
           }
       }
       return newdic;
   }
}  // namespace etl
